src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```
@@ -0,0 +1,72 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(
+                    "Can't add movieSession entity, error: ", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movieSession;
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> movieSessionQuery = session.createQuery("from MovieSession m "
+                    + "where m.id = :id", MovieSession.class);
+            movieSessionQuery.setParameter("id", id);
+            return Optional.ofNullable(movieSessionQuery.getSingleResult());

```

This implementation uses getSingleResult() and wraps the result with Optional.ofNullable(...). However, getSingleResult() will throw NoResultException when there is no matching row, so the method will throw instead of returning Optional.empty().

This violates checklist item 2.3: 'DAO get(...) methods should return Optional<T> rather than null or throwing if not found (see checklist item 6).' and checklist item 2.7: 'When querying single entity by id at DAO layer, use an appropriate API for single/unique results; prefer uniqueResultOptional() or an equivalent approach that safely returns Optional (see checklist item 7).'

Recommendation: replace the query/get logic with either:
- session.get(MovieSession.class, id) and return Optional.ofNullable(...), or
- a query that calls uniqueResultOptional() so absence maps to Optional.empty() without throwing.